src/sentry/replays/consumers/recording.py

```diff
@@ -188,14 +216,31 @@ def parse_headers(recording: bytes, replay_id: str) -> tuple[int, bytes]:
 # I/O Task
 
 
-def commit_message_with_options(message: Message[ProcessedEvent]) -> None:
+def commit_message_with_profiling(message: Message[ProcessedEvent]) -> None:
     profiling_enabled = options.get(
         "replay.consumer.recording.profiling.enabled",
     )
-    return commit_message(message, profiling_enabled=profiling_enabled)
 
+    if profiling_enabled:
```

Comment:
we might be able to consolidate `profiling_enabled` and the `SENTRY_REPLAY_CONSUMER_PROFILING_SAMPLE_RATE`

[these docs](https://docs.sentry.io/pricing/quotas/manage-continuous-profile-hours/#how-can-i-control-when-profiling-is-active)  say that:
>Note that a profile session has to be sampled based on the value of profile_session_sample_rate for start_profiler to have any effect. Calling start_profile_session will have no effect on un-sampled sessions.